src/controllers/submitUrlController.js

```diff
@@ -58,6 +65,52 @@ class SubmitUrlController extends UploadController {
   static urlIsNotTooLong (url) {
     return url.length <= 2048
   }
+
+  static async getHeadRequest (url) {
+    try {
+      return await axios.head(url)
+    } catch (err) {
+      console.error(err)
+      return null
+    }
+  }
+
+  static urlResponseIsNotTooLarge (response) {
+    try {
+      const contentLength = response.headers['content-length']
+
+      // Convert content length to MB
+      const sizeInMB = contentLength / (1024 * 1024)
+
+      return sizeInMB <= 10
+    } catch (err) {
+      console.error(err)
+      return true // for now we will allow this file as we can't be sure
```

Comment:
How do we actually reach this catch block (and for `urlExists`) ie what kind of errors could happen here? There aren't test cases to describe how we get there and it seems risky to allow things to pass validation (`return true`) without really understanding what we're catching

Reply:
if no response is returned, or if the response doesn't have a content-length property
and this is only local validation, to do basic checks before sending the url to the backend where more in depth checks are done. so we want to be more forgiving here.

Reply:
Okay fair tbh this was me thinking javascript was way more forgiving than it is and would be happy trying to read `.headers` on `undefined`